hathor/transaction/base_transaction.py

```diff
@@ -886,8 +897,12 @@ def update_accumulated_weight(self, *, stop_value: float = inf, save_file: bool
 
         return metadata
 
-    def update_initial_metadata(self) -> None:
-        """Update the tx's initial metadata. It does not update the whole metadata.
+    def update_reward_lock_metadata(self) -> None:
+        metadata = self.get_metadata()
+        metadata.min_height = self.calculate_min_height()
```

Comment:
You have to save it, right?

Reply:
Maybe, it depends on when we use it. On this implementation we update this metadata _before_ validating, we don't want to save a potentially invalid transaction/block.